src/stdatamodels/jwst/transforms/models.py

```diff
@@ -1989,6 +1991,158 @@ def _evaluate_transform_guess_form(model, x=0, y=0, t=0):
+        """
+        wavelength = np.atleast_1d(wavelength)
+        x = np.atleast_1d(x)
+        y = np.atleast_1d(y)
+
+        if (wavelength < 0).any():
+            raise ValueError("Wavelength should be greater than zero")
+        try:
+            iorder = self._order_mapping[int(order.flatten()[0])]
+        except KeyError as err:
+            raise ValueError("Specified order is not available") from err
+        # t is trace normalization parameters (it was values of 0 to 1)
+
+        t = self.lmodels[iorder](wavelength)
+        xmodel = self.xmodels[iorder]
+        ymodel = self.ymodels[iorder]
+        dx = xmodel(t)
+        dy = ymodel[0](x, y) + dx * ymodel[1](x, y) + dx**2 * ymodel[2](x, y)
+        return x + dx, dy, x, y, order
+
+
+class MIRIWFSSForwardDispersion(_ForwardGrismDispersionBase):
+    """
+    Calculate the wavelengths of vertically dispersed MIRIWFSS data.
+
+    The dispersion polynomial is relative to the input x,y pixels
+    in the direct image for a given wavelength.
+    """
+
+    def __init__(self, orders, lmodels=None, xmodels=None, ymodels=None, name=None):
+        """
+        Initialize the model.
+
+        Parameters
+        ----------
+        orders : list
+            The list of orders which are available to the model
+        lmodels : list
+            The list of models for the polynomial model in l
+        xmodels : list[tuples]
+            The list of tuple(models) for the polynomial model in x
+        ymodels : list[tuples]
+            The list of tuple(models) for the polynomial model in y
+            Name of the model
+        """
+        if name is None:
+            name = "miri_wfss_forward_dispersion"
+        super().__init__(
+            orders,
+            lmodels=lmodels,
+            xmodels=xmodels,
+            ymodels=ymodels,
+            name=name,
+        )
+
+    def evaluate(self, x, y, x0, y0, order):
+        """
+        Transform from the dispersed plane into the direct image plane.
+
+        Parameters
+        ----------
+        x, y :  int, float, list
+            Input x, y location in the dispersed image.
+        x0, y0 : int, float, list
+            Source object x-center, y-center in the direct image.
+        order : int
+            Input spectral order
+
+        Returns
+        -------
+        x, y : float
+            The x, y values in the direct image, same as x0, y0.
+        lambda : float
+            Wavelength in microns
+        order : int
+            Output spectral order, same as input
+        """
+        try:
+            iorder = self._order_mapping[int(order.flatten()[0])]
+        except KeyError as err:
+            raise ValueError("Specified order is not available") from err
+
+        # The next two lines are to get around the fact that
+        # modeling.standard_broadcasting=False does not work.
+        x00 = x0.flatten()[0]
+        y00 = y0.flatten()[0]
+
+        t = np.linspace(0, 1, 10)  # sample t
```

Comment:
```suggestion
        t = np.linspace(0, 1, self.sampling)  # sample t
```
Instead of hard-coding this to have 10 values, can you use the `self.sampling` attribute instead?  (`_GrismDispersionBase` already has this attribute defined and set to a default value of 40, which I assume would also work fine here.  But if you want to change the default you can always override it in this class)

Reply:
Ok yes I can do that.